source/common/http/header_map_impl.cc

```diff
@@ -304,6 +293,22 @@ HeaderMapImpl::HeaderMapImpl(
   }
 }
 
+void HeaderMapImpl::copyFrom(const HeaderMap& header_map) {
+  header_map.iterate(
```

Comment:
If you are defining a copy-ctor I think you should also define an assignment operator (and test it).

https://en.cppreference.com/w/cpp/language/rule_of_three

I see you are defining one in a test subclass, but it doesn't look like you've got one in HeaderMapImpl.

Reply:
Yes, that's fair. I will mvoe copy assignment to `HeaderMapImpl`.